docs/evaluation/load_test_endpoint.Rmd

```diff
@@ -0,0 +1,125 @@
+---
+title: "LoadTest"
+author: "Sebastian Garzon"
+output: md_document
+---
+
+This documet contais some script to test the o2r API endpoint using the `loadtest` [library](https://github.com/tmobile/loadtest).
+
+## Required libraries
+```{r libs}
+library("loadtest")
+library("ggplot2")
+library("dplyr")
+library("tidyr")
+```
+
+## Other requirements
+
+In order to use the `loadtest` library it is required to install Java and Jmeter. Other requirements are described on the [installation document](https://github.com/tmobile/loadtest#installation) of this library.
+
+## Preparation
+
+This documet contais some script to test the o2r API.
+
+**Note:** This document is best run chunk by chunk, and not as a whole, using RStudio.
+`loadtest` library requires the path to the jmeter bin folder to work. You can add a file `.Renviron` on your local machine next to this file, and it define the path in the environment variable "LOADTEST_JMETER_PATH". For more information about this local variable refer to this [guide](https://github.com/tmobile/loadtest#installing-jmeter).
+
+```{r  cache=TRUE,message=FALSE}
+
+#Building DataSets
+
+#Configuration required to use 'loadtest' library, this system variable is stored in a .Renviron file. LOADTEST_JMETER_PATH="[path to jmeter bin folder]"
+
+#Sys.getenv("LOADTEST_JMETER_PATH")
+
+```
+# Test 
+
+## LoadTest o2r API entry point
+
+The purpose of this test is to evaluate the performance of the o2r API entry point simulating users making multiple request simultaneously. Every test is going to be defined by a number of users (Threads) and the number of requests per user (same number for every user). The `test_threads` list define the number of users for test and `request_num` the number of request per user. For example with `test_threads = c(1,2)` and `request_num =500` there are going to be 2 tests, the first with one thread of 500 requests and the second one with two threads of 500 requests per thread (i.e., 1000 request in total).
+
+```{r dataProcess, eval=FALSE, cache=TRUE,message=FALSE}
+
+# Define test
+test_threads<-c(1,2,3,4,5,6)
+# Define individual number of request
+request_num<-2000
+
+# List with the number of loops for each test scenario (per Thread). 
+
+number_loops<-rep.int(request_num,length(test_threads))
+
+# Run multiple consecutive tests
+
+test_variables  <- paste("Test_threads_",test_threads,sep="")
+test_functions<-paste0('loadtest(url = "https://o2r.uni-muenster.de/api/v1", method = "GET", threads =',test_threads,', loops=',number_loops,')')
+eq<-paste(paste(test_variables,test_functions,sep="<-"),collapse = ";")
+eval(parse(text=eq))
+
+#Save image of the current R environment
+
+save.image(file='datasets.RData')
```

Comment:
For reproducibility, never use use `save.image()`. Save the data in a plain text format, such as CSV.